core/classes/TimelineEvent.class.php

```diff
@@ -80,9 +80,9 @@ public function html_start() {
 
 		return sprintf(
 			'<div class="entry"><div class="avatar"><a href="%s"><img class="avatar" src="%s" alt="%s" width="32" height="32" /></a></div><div class="timestamp">%s</div>',
-			$t_avatar->link,
-			$t_avatar->image,
-			$t_avatar->text,
+			htmlspecialchars( $t_avatar->link ),
+			htmlspecialchars( $t_avatar->image ),
+			htmlspecialchars( $t_avatar->text ),
 			$this->format_timestamp( $this->timestamp )
```

Comment:
I'm wondering if we need to htmlspecialchars() the timestamp too.

This is not consistently done in the code, but sometimes we call string_display_line() after formatting the date, and sometimes not.